docs/fsharp/tour.md

```diff
@@ -1,4 +1,158 @@
+
+[[!code-fsharp[PatternMatching](../../samples/snippets/fsharp/tour.fs#L530-L547)]
+
+Something you may have noticed is the use of the `_` pattern.  This is known as the "wildcard" pattern, which is basically a way of saying "I don't care what it is, since it doesn't matter to me".  Although convenient, you can accidentaly bypass Exhaustive Pattern Matching and no longer benefit from compile-time enforcements if you aren't careful.
+
+[Active Patterns](language-reference/active-patterns.md) are another powerful construct to use with pattern matching.  They allow you to partition input data into custom forms, decomposing them at the pattern match call site.  They can also be parameterized, thus allowing to define the partition as a function.  Expanding the previous example to support Active Patterns looks something like this:
+
+[[!code-fsharp[ActivePatterns](../../samples/snippets/fsharp/tour.fs#L530-L564)]
+
 >![NOTE]
-This is in-progress.
\ No newline at end of file
+The previous above example creates a `Result` type.  This has been added to the language with F# 4.1.  This document will be updated once F# 4.1 is shipped.
+
+## Optional Types
+
+One special case of Discriminated Union types is the Option Type, which is so useful that it's a part of the F# core library.
+
+[The Option Type](language-reference/options.md) is a type which represents one of two cases: a value, or nothing at all.  It is used in any scenario where a value may or may not result from a particular operation.  This then forces you to account for both cases, making it a compile-time concern rather than a runtime concern.  These are often used in APIs where `null` is used to represent "nothing" instead, thus eliminating the need to worry about `NulReferenceException` in many circumstances.
+
+[[!code-fsharp[Options](../../samples/snippets/fsharp/tour.fs#L472-L489)]
+
+## Units of Measure
+
+One unique feature of F#'s type system is the ability to provide context for numeric literals through Units of Measure.
+
+[Units of Measure](language-reference/units-of-measure.md) allow you to associate a numeric type to a unit, such as Meters, and have functions perform work on units rather than numeric literals.  This enables the compiler to verify that the types of numeric literals passed in make sense under a certain context, and eliminate runtime errors associated with that kind of work.
+
+[[!code-fsharp[UnitsOfMeasure](../../samples/snippets/fsharp/tour.fs#L570-L585)]
+
+The F# Core library defines many SI unit types and unit conversions.  To learn more, check out the [Microsoft.FSharp.Data.UnitSystems.SI Namespace](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/microsoft.fsharp.data.unitsystems.si-namespace-%5bfsharp%5d).
+
+## Classes and Interfaces
```

Comment:
As a reader, I'd want to see a little guidance based on these topics: Are these techniques used primarily for interop with other .NET languages? Do they have merit in their own right? Should I prefer the functions and modules (introduced above) to these techniques?

I think a short paragraph would be sufficient.